### ISSUE TITLE
Property 'fill' does not exist on type 'Runner'.

### ISSUE THREAD

**Issue by twoco** [svgdotjs/svg.js#1117]
Example used from [Getting Started](https://svgjs.com/docs/3.0/getting-started/) docs.

```ts
const draw = SVG().addTo('.draw').size('100%', '100%');
draw.rect(100, 100).animate().fill('#f03').move(100, 100);
```

![image](https://user-images.githubusercontent.com/53160891/83073163-42758200-a070-11ea-8628-9a249c7429d7.png)

But on runtime it works. It's just a type issue.

```json
{
  "@svgdotjs/svg.js": "^3.0.16",
  "typescript": "~3.8.3",
}
```

**Comment by Fuzzyma** [comment]
Yes, guess the typings for the runner are just not complete. If you want you can edit the file directly in your browser and submit a PR :)

**Comment by twoco** [comment]
I see that there is one `d.ts` file for the types. Crafted by hand, right? Not automatically generated from the sources. ...

The class `Runner` should extends or implements from `Sugar`? It's hard to follow / find the assigned methods. I see that the methods are dynamically added by `addMethodNames` / `registerMethods` etc. No offence but this is not clean code. Concider to use TypeScript with strict types to prevent / avoid "hacks" (magic JavaScript stuff 😉 ) like this. Another advantage is automatic generated definition files (`d.ts`). Doc comments will also help.
Just a well-intentioned advice. But it's good that you use modern JavaScript. 👍

**Comment by saivan** [comment]
I don’t take offence no. But you must realise that this library is a library. These aren’t hacks, they are designed to allow svgjs to be extensible and compose-able, these were design decisions made for a good reason. You can use svgjs without the sugar by editing the main file which wouldn’t work if we went down the route of extending everything, you’d end up with hundreds of files to edit. Typescript support is secondary to that for us. 

This library is quite old and the definition of clean code changes very often and we don’t want to try keeping up. I appreciate your intentions, and I’m thankful for your contributions. But before we start waving our hack stick, we should ask why this design was chosen first.
